contracts/src/v0.7/dev/KeeperRegistryDev.sol

```diff
@@ -355,6 +355,20 @@ contract KeeperRegistryDev is
     LINK.transfer(to, amount);
   }
 
+  /**
+   * @notice allows the admin of an upkeep to modify gas limit
+   * @param id upkeep to be change the gas limit for
+   * @param gasLimit new gas limit for the upkeep
+   */
+  function setGasLimit(uint256 id, uint32 gasLimit) external override onlyActiveUpkeep(id) onlyUpkeepAdmin(id) {
+    require(gasLimit >= CALL_GAS_MIN, "min gas is 2300");
+    require(gasLimit <= CALL_GAS_MAX, "max gas is 5000000");
```

Comment:
I think we should probably convert `CALL_GAS_MAX` to another config variable. It's currently assumed that once we accept things from the registrar that they're immutable. This obviously changes that, but making the upper bound more configurable would allow us to adjust the conservative upper bound.

Reply:
Sounds good, I'm thinking of only allowing setting the max limit in constructor, because if owner changes it later on, it may result in upkeeps violating the gas limit. 

Also @RyanRHall during migration we will need to check the upkeep's gas limit satisfies the registry's requirement

Reply:
@infiloop2 good call out! I think I have a decent solution for that.